include/rootfs.mk

```diff
@@ -78,7 +78,13 @@ define prepare_rootfs
 		done; \
 		for script in ./etc/init.d/*; do \
 			grep '#!/bin/sh /etc/rc.common' $$script >/dev/null || continue; \
-			IPKG_INSTROOT=$(1) $$(which bash) ./etc/rc.common $$script enable; \
+			if ! echo " $(3) " | grep -q " $$(basename $$script) "; then \
```

Comment:
I'm not sure if this approach is going to work properly in all corner cases, as for example this made up one `DISABLED_SERVICES=httpd` would disable as well all other services with `httpd` in it's name, so `uhttpd`, `microhttpd` or `lighttpd`.

Reply:
It should work thanks to the "spacing" added around the grepped expanded service name, so the grep should match only the "whole words". The matched list ($(3)) gets also expanded by spaces at the beginning and at the end to allow also the first and the last service in the list to match (if needed).

Here is an example to demonstrate clearly the idea:
```
#!/bin/bash
SERVICES="httpd uhttpd microhttpd lighthttpd"
DISABLE="httpd"

for s in $SERVICES; do
    if ! echo " $DISABLE " | grep -q " $s "; then
        echo "Enabling" $s;
    else
        echo "Disabling" $s;
    fi;
done;
```